apps/qubit/modules/object/actions/exportAction.class.php

```diff
@@ -0,0 +1,159 @@
+        return parent::addField($name);
+    }
+  }
+
+  protected function processField($field)
+  {
+    switch ($field->getName())
+    {
+      case 'levels':
+        $this->levels = $this->form->getValue('levels');
+        if (empty($this->levels))
+        {
+          $this->levels = array();
+        }
+
+        break;
+    }
+  }
+
+  protected function doBackgroundExport($request)
+  {
+    // Export type, CSV or XML?
+    $exportType = $request->getParameter('format', 'xml');
+
+    // We will use this later to redirect users back to the importSelect page
+    if (isset($this->getRoute()->resource))
+    {
+      $exportRoute = array($this->getRoute()->resource, 'module' => 'object', 'action' => 'importSelect', 'type' => $exportType);
+    }
+    else
+    {
+      $exportRoute = array('module' => 'object', 'action' => 'export', 'type' => $exportType);
+    }
+
+    // Create array of selections to pass to background job where Term ID will
+    // be key, and Term description is value.
+    foreach($this->levels as $key => $value)
+    {
+      $levelsOfDescription[$value] = $this->choices[$value];
+    }
+
+    $options = array('params' => array('fromClipboard' => true,
+                                       'slugs' => $this->context->user->getClipboard()->getAll()),
+                     'include-all-levels' => ('on' == $request->getParameter('includeAllLevels')) ? true : false,
+                     'current-level-only' => ('on' == $request->getParameter('includeDescendants')) ? false : true,
+                     'public' => ($request->getParameter('includeDrafts') == 'on') ? false : true,
+                     'objectType' => $request->getParameter('objectType'),
+                     'levels' => $levelsOfDescription);
+
+    try
+    {
+      if ('CSV' == strtoupper($exportType))
+      {
+        QubitJob::runJob('arInformationObjectCsvExportJob', $options);
+      }
+      else
+      {
+        QubitJob::runJob('arXmlExportJob', $options);
+      }
+
+      // Let user know import has started
+      sfContext::getInstance()->getConfiguration()->loadHelpers(array('Url'));
+      $jobManageUrl = url_for(array('module' => 'jobs', 'action' => 'browse'));
+      $message = '<strong>Export of descriptions initiated.</strong> Check <a href="'. $jobManageUrl . '">job management</a> page to download the results when it has completed.';
+      $this->context->user->setFlash('notice', $this->context->i18n->__($message));
+    }
+    catch (sfException $e)
+    {
+      $this->context->user->setFlash('error', $e->getMessage());
+      $this->redirect($exportRoute);
+    }
+  }
+
+  public function execute($request)
+  {
+    parent::execute($request);
+
+    if ($request->isMethod('post'))
+    {
+      $this->form->bind($request->getPostParameters());
+
+      if ($this->form->isValid())
+      {
+        $this->processForm();
+
+        $this->doBackgroundExport($request);
+
+        $this->setTemplate('exportResults');
```

Comment:
If it reaches the end of the execute method without a redirect or a return it will load the default template, the same as doing `return sfView::SUCCESS` at the end